e2e/pipelines/test_evaluation_pipeline.py

```diff
@@ -0,0 +1,273 @@
+    ingestion_pipe.add_component(instance=doc_writer, name="doc_writer")
+    ingestion_pipe.connect("doc_embedder.documents", "doc_writer.documents")
+    ingestion_pipe.run({"doc_embedder": {"documents": documents}})
+    return document_store
+
+
+def rag_pipeline(document_store: InMemoryDocumentStore, top_k: int):
+    """RAG pipeline"""
+    template = """
+        You have to answer the following question based on the given context information only.
+
+        Context:
+        {% for document in documents %}
+            {{ document.content }}
+        {% endfor %}
+
+        Question: {{question}}
+        Answer:
+        """
+    rag = Pipeline()
+    rag.add_component("embedder", SentenceTransformersTextEmbedder(model=embeddings_model, progress_bar=False))
+    rag.add_component("retriever", InMemoryEmbeddingRetriever(document_store, top_k=top_k))
+    rag.add_component("prompt_builder", PromptBuilder(template=template))
+    rag.add_component("generator", OpenAIGenerator(model="gpt-3.5-turbo"))
+    rag.add_component("answer_builder", AnswerBuilder())
+    rag.connect("embedder", "retriever.query_embedding")
+    rag.connect("retriever", "prompt_builder.documents")
+    rag.connect("prompt_builder", "generator")
+    rag.connect("generator.replies", "answer_builder.replies")
+    rag.connect("generator.meta", "answer_builder.meta")
+    rag.connect("retriever", "answer_builder.documents")
+
+    return rag
+
+
+def eval_pipeline(questions, truth_docs, truth_answers, retrieved_docs, contexts, pred_answers):
+    """
+    Run the evaluation pipeline
+    """
+    eval_pipeline = Pipeline()
+    eval_pipeline.add_component("doc_mrr", DocumentMRREvaluator())
+    eval_pipeline.add_component("groundness", FaithfulnessEvaluator())
+    eval_pipeline.add_component("sas", SASEvaluator(model=embeddings_model))
+    eval_pipeline.add_component("doc_map", DocumentMAPEvaluator())
+    eval_pipeline.add_component("doc_recall_single_hit", DocumentRecallEvaluator(mode=RecallMode.SINGLE_HIT))
+    eval_pipeline.add_component("doc_recall_multi_hit", DocumentRecallEvaluator(mode=RecallMode.MULTI_HIT))
+    eval_pipeline.add_component("relevance", ContextRelevanceEvaluator())
+
+    return eval_pipeline.run(
+        {
+            "doc_mrr": {"ground_truth_documents": truth_docs, "retrieved_documents": retrieved_docs},
+            "groundness": {"questions": questions, "contexts": contexts, "responses": truth_answers},
+            "sas": {"predicted_answers": pred_answers, "ground_truth_answers": truth_answers},
+            "doc_map": {"ground_truth_documents": truth_docs, "retrieved_documents": retrieved_docs},
+            "doc_recall_single_hit": {"ground_truth_documents": truth_docs, "retrieved_documents": retrieved_docs},
+            "doc_recall_multi_hit": {"ground_truth_documents": truth_docs, "retrieved_documents": retrieved_docs},
+            "relevance": {"questions": questions, "contexts": contexts},
+        }
+    )
```

Comment:
We should move this to another function as the eval pipeline only needs to be created once.

Reply:
good point, one less big object creation 👍🏽